routes/warehouses.js

```diff
@@ -6,19 +6,21 @@ const router = express.Router();
 const { v4: uuid } = require('uuid');
 const { validatePhone, validateEmail } = require('../middleware/validators');
 
-// API to GET List of All warehouses
+// GET List of All warehouses
 router.get("/", function (req, res) {
     res.status(201).json(warehouseData)
 })
 
-// API to GET List of Single warehouses
+// GET single warehouse and inventory list
 router.get("/:id", function (req, res) {
     
     const paramsID = req.params.id
     const singleWarehouse = warehouseData.find(warehouse => warehouse.id === paramsID)
+    const warehouseInventory = inventoryData.filter(inventory => inventory.warehouseID === singleWarehouse.id )
```

Comment:
If we try a the route with a bad ID, it gets an error because `singleWarehouse` is undefined. Because of this, it will also never reach line 26 for `res.status(400)`

Something similar to the following:
```
TypeError: Cannot read properties of undefined (reading 'id')
    at C:\Users\tli92\Desktop\BrainStation\instock\instock-api\routes\warehouses.js:19:108
    at Array.filter (<anonymous>)
    at C:\Users\tli92\Desktop\BrainStation\instock\instock-api\routes\warehouses.js:19:46
    at Layer.handle [as handle_request] (C:\Users\tli92\Desktop\BrainStation\instock\instock-api\node_modules\express\lib\router\layer.js:95:5)
    at next (C:\Users\tli92\Desktop\BrainStation\instock\instock-api\node_modules\express\lib\router\route.js:144:13)
    at Route.dispatch (C:\Users\tli92\Desktop\BrainStation\instock\instock-api\node_modules\express\lib\router\route.js:114:3)
    at Layer.handle [as handle_request] (C:\Users\tli92\Desktop\BrainStation\instock\instock-api\node_modules\express\lib\router\layer.js:95:5)
    at C:\Users\tli92\Desktop\BrainStation\instock\instock-api\node_modules\express\lib\router\index.js:284:15
    at param (C:\Users\tli92\Desktop\BrainStation\instock\instock-api\node_modules\express\lib\router\index.js:365:14)
    at param (C:\Users\tli92\Desktop\BrainStation\instock\instock-api\node_modules\express\lib\router\index.js:376:14)
```